R/utilities-simulation.R

```diff
@@ -180,26 +180,22 @@ simulateModelParallel <- function(structureSets,
       pathFolder = logFolder,
       logTypes = LogTypes$Info
     )
-    
-    # Catch, save and display any error or warning
-    # from ospsuite::runSimulations
-    subsetSimulationResults <- tryCatch({
-       ospsuite::runSimulations(simulations = simulations)
-    },
-    error = function(e) {
-      logErrorThenStop(message = e, logFolder)
-      },
-    warning = function(w) {
-      logWorkflow(
-        message = w,
-        pathFolder = logFolder,
-        logTypes = LogTypes$Error
-      )
-      # Since simulationResults is a list,
-      # return same output type
-      return(vector(mode = "list", length = length(simulations)))
-    })
-    
+
+    # Stop simulation run if any simulations not run succesfully in ospsuite::runSimulations
+    subsetSimulationResults <-  ospsuite::runSimulations(simulations = simulations)
+    failedSimulationIndices <- which( sapply(subsetSimulationResults,function(x){is.null(x)}) )
+
+    if(!isOfLength(failedSimulationIndices,0)){
```

Comment:
@abdullahhamadeh 
We have added a new wrapper in `{ospsuite.utils}` called `isEmpty`.
You can use it instead of `isOfLength(.., 0)` (and you can also use it instead of `nrow(...)=0`)